### ISSUE TITLE
Stop using directly classes from dao-api to generate download files

### ISSUE THREAD

**Issue by fbastian** [BgeeDB/bgee_apps#12]
In GitLab by @fbastian on Jul 14, 2015, 14:23

In `org.bgee.pipeline.expression.downloadfile`, the classes generating download files use the class `org.bgee.model.dao.api.expressiondata.CallDAO.CallTO.DataState` to generate the data state column of the download files. This is problematic, as the String representation from `bgee-dao-api` is not guaranteed to correspond to the expected values in download files.

There should be no use of `org.bgee.model.dao.api.expressiondata.CallDAO.CallTO.DataState` to generate values to display in download files.

See also issue #8 about the use of the method `org.bgee.pipeline.expression.downloadfile.GenerateDiffExprFile.convertDataStateToString(DataState)`

Also, we use directly the class `org.bgee.model.dao.api.expressiondata.DiffExpressionCallDAO.DiffExpressionCallTO.ComparisonFactor`, see for instance `org.bgee.pipeline.expression.downloadfile.GenerateMultiSpeciesDiffExprFile.MultiSpeciesDiffExprFileType`. 

Basically, we should never use directly classes from `org.bgee.model.dao.api` for this kind of uses.
